include/boost/beast/core/detail/impl/temporary_buffer.ipp

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright (c) 2019 Damian Jarek(damian.jarek93@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_DETAIL_IMPL_TEMPORARY_BUFFER_IPP
+#define BOOST_BEAST_DETAIL_IMPL_TEMPORARY_BUFFER_IPP
+
+#include <boost/beast/core/detail/temporary_buffer.hpp>
+#include <boost/beast/core/detail/clamp.hpp>
+#include <boost/core/exchange.hpp>
+
+#include <memory>
+#include <cstring>
+
+namespace boost {
+namespace beast {
+namespace detail {
+
+
+void
+temporary_buffer::append(string_view sv)
+{
+    reserve_additional(sv.size());
+    unsafe_append(sv);
+}
+
+void
+temporary_buffer::append(string_view sv1, string_view sv2)
+{
+    reserve_additional(sv1.size() + sv2.size());
+    unsafe_append(sv1);
+    unsafe_append(sv2);
+}
+
+void
+temporary_buffer::unsafe_append(string_view sv)
```

Comment:
I prefer the term `unchecked_append`. This is a marketing issue: I rather not have people saying that beast has "unsafe" functions. But this is a minor nitpick.